spring-boot-email-core/src/main/java/it/ozimov/springboot/mail/service/defaultimpl/DefaultEmailService.java

```diff
@@ -123,7 +124,7 @@ public MimeMessage send(final @NonNull Email email,
                 DataSource source = new ByteArrayDataSource(emailAttachment.getAttachmentData(),
                         emailAttachment.getContentType().toString());
                 attachmentPart.setDataHandler(new DataHandler(source));
-                attachmentPart.setFileName(emailAttachment.getAttachmentName());
+                attachmentPart.setFileName(MimeUtility.encodeText(emailAttachment.getAttachmentName()));
```

Comment:
As far as the doc says, only headers must be MIME encoded http://docs.oracle.com/javaee/6/api/javax/mail/internet/MimeUtility.html#encodeText(java.lang.String)

Read also: https://stackoverflow.com/questions/28452599/javamail-attachment-filename-not-displaying-utf-8-characters-correctly

Anyway, if we want to go for this solution, then we need to set the same encoding as the email.
``Email`` class has ``getEncoding()``.

Maybe it should be ``MimeUtility.encodeText(emailAttachment.getAttachmentName(), email.getEncoding(), null)``;
But unit tests should confirm the former or the latter approach.